src/abstractarray.jl

```diff
@@ -154,6 +154,7 @@ reshape(a::Array, s::Size{S}) where {S} = s(a)
 @inline vec(a::StaticArray) = reshape(a, Size(prod(Size(typeof(a)))))
 
 @inline copy(a::StaticArray) = typeof(a)(Tuple(a))
+@inline copy(a::SizedArray) = typeof(a)(a)
```

Comment:
Does this really create a copy?

Reply:
As far as I can tell:
```julia
julia> copy(a::SizedArray) = typeof(a)(a)
copy (generic function with 88 methods)

julia> a = Size(100)(rand(100));

julia> b = copy(a);

julia> b[1] = 2
2

julia> a[1], b[1]
(0.3688743314251053, 2.0)
```

Reply:
Another option would be to make `SizedArray` subtype of `AbstractArray` rather than `StaticArray`.

Reply:
@andyferris Another option would be to make `SizedArray` subtype of `AbstractArray` rather than `StaticArray`.

Reply:
Uh oh. You're right @BeastyBlacksmith, this is creating a copy:

```
julia> using StaticArrays, Test

julia> v = SizedVector{2}(zeros(2))
2-element SizedArray{Tuple{2},Float64,1,1}:
 0.0
 0.0

julia> v2 = SizedVector{2}(v)
2-element SizedArray{Tuple{2},Float64,1,1}:
 0.0
 0.0

julia> @which SizedVector{2}(v)
(::Type{SA})(a::StaticArray) where SA<:StaticArray in StaticArrays at ~/.julia/dev/StaticArrays/src/convert.jl:5

julia> @test v2.data === v.data
Test Failed at REPL[16]:1
  Expression: v2.data === v.data
   Evaluated: [0.0, 0.0] === [0.0, 0.0]
ERROR: There was an error during testing
```

But I think this is a bug; I believe the intention of `SizedArray` was to wrap the underlying data. Indeed, we have

```julia
julia> @test SizedVector{2}(data).data === data
Test Passed
```

which seems rather inconsistent as the normal constructor simply wraps the data rather than copies it.

I think the correct solution here is to add a `copy(a)` to your implementation, and to specialize the constructor of `SizedArray` (or, possibly better, `convert`?) so that it wraps rather than copies the data.